src/frontend/src/lib/components/send/SendInputAmount.svelte

```diff
@@ -0,0 +1,62 @@
+<script lang="ts">
+	import { Input } from '@dfinity/gix-components';
+	import { debounce, nonNullish } from '@dfinity/utils';
+	import { parseToken } from '$lib/utils/parse.utils';
+	import { slide } from 'svelte/transition';
+	import { i18n } from '$lib/stores/i18n.store';
+	import type { BigNumber } from '@ethersproject/bignumber';
+	import { invalidAmount } from '$lib/utils/input.utils';
+	import MaxButton from '$lib/components/common/MaxButton.svelte';
+
+	export let amount: number | undefined = undefined;
+	export let tokenDecimals: number | undefined = undefined;
+	export let placeholder: string = $i18n.core.text.amount;
+	export let customValidate: (userAmount: BigNumber) => Error | undefined = () => undefined;
+	export let calculateMax: (() => number | undefined) | null = null;
+
+	let error: Error | undefined;
+
+	let onMax = () => {
+		amount = calculateMax?.();
+	};
+
+	const validate = () => {
+		error = undefined;
```

Comment:
nth: I would set `error = undefined` only if required. That way you avoid a brief rerender if the error is set, then undefined, then set again.

Long story short:

```
const validate = () => {
   if (invalidAmount(amount)) {
        error = undefined;
        return;
   }
   
   ...
}
```